### ISSUE TITLE
remote error: tls: bad certificate

### ISSUE THREAD

**Issue by mf-lit** [kubernetes/helm#4114]
I'm relatively new to k8s and helm, but have been happily running with both for a few weeks now. I use TLS to secure my helm-tiller connection.

Yesterday, I did a kops rolling-update on my cluster (in AWS) in order to increase the instance size of my nodes, which caused a few niggles but fundamentally worked, all my resources remained on-line and are operational. But suddenly my helm client can't communicate with tiller over tls.
```
$  helm list --tls --debug
[debug] Created tunnel using local port: '33277'

[debug] SERVER: "127.0.0.1:33277"

[debug] Key="/home/marc/.helm/key.pem", Cert="/home/marc/.helm/cert.pem", CA="/home/marc/.helm/ca.pem"

Error: remote error: tls: bad certificate
```

I've checked that the certificates in $HELM_HOME haven't changed, I've also diff'd my original copies of the tiller and CA certificates against those stored in the secret, and confirmed that they haven't changed.

The secret itself hasn't changed at all, but of course the tiller pod has been terminated and recreated as part of the cluster upgrade, so I assume this _must_ have something to do with it. I've tried deleting the tiller pod manually and letting it recreate, but it makes no difference. The logs for the tiller pod show nothing interesting at all.

Any guidance on what to do next please? I've no idea how to (a) fix it (b) provide more useful diagnostics for this issue.

Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Error: cannot connect to Tiller
```
Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.2", GitCommit:"81753b10df112992bf51bbc2c2f85208aad78335", GitTreeState:"clean", BuildDate:"2018-04-27T09:22:21Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T11:55:20Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```
Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
AWS/Kops

**Comment by mf-lit** [comment]
So I'm an idiot, my tiller and helm certs happened to have expired on the same day that I did this rolling-update.

To help fellow idiots, I've added some more details to the TLS guide in the docs (as the existing suggested commands for signing certs default to just 30 days)

https://github.com/kubernetes/helm/pull/4121

**Comment by bacongobbler** [comment]
closed via #4121
